core/src/io/anuke/mindustry/world/blocks/power/PowerGraph.java

```diff
@@ -39,37 +38,41 @@ public synchronized void update(){
         lastFrameUpdated = threads.getFrameID();
 
         float totalInput = 0f;
-
         for(Tile producer : producers){
-            totalInput += producer.entity.power.amount;
+            float accumulator = producer.entity.power.amount;
+            if (accumulator > 0.0001f) {
```

Comment:
The general logic seems to be fine here, but most of the if-statements and extra variable declarations are redundant. `totalInput += producer.entity.power.amount` should happen regardless of whether `accumulator > 0.0001f`.

I suggest replacing the entire contents of this for-loop with just `totalInput += producer.entity.power.amount;`.